simple/rdm_triggeredOps.c

```diff
@@ -0,0 +1,670 @@
+				 tx_buf + strlen(welcome_text),
+				 strlen(welcome_text), &work.context, FI_UINT8, FI_BOR);
+	format_simple_msg_fetch(&fetch, &fetch_iov, result_buf, strlen(welcome_text));
+
+	fetch_atomic.ep = trig_ep;
+	fetch_atomic.msg = msg;
+	fetch_atomic.fetch = fetch;
+	fetch_atomic.flags = 0;
+	work.op_type = FI_OP_FETCH_ATOMIC;
+
+	work.op.fetch_atomic = &fetch_atomic;
+	return fi_control(&domain->fid, FI_QUEUE_WORK, &work);
+}
+
+static int compare_atomic_trigger()
+{
+	int ret;
+	size_t count;
+	struct fi_op_compare_atomic compare_atomic;
+	struct fi_msg_atomic msg;
+	struct fi_msg_fetch fetch;
+	struct fi_msg_compare compare;
+	struct fi_ioc iov;
+	struct fi_ioc fetch_iov;
+	struct fi_ioc compare_iov;
+	struct fi_rma_ioc rma_iov;
+
+	ret = check_compare_atomic_op(ep, FI_CSWAP, FI_UINT8, &count);
+	if (ret)
+		return ret;
+
+	format_simple_msg_atomic(&msg, &iov, &rma_iov,
+				 tx_buf + strlen(welcome_text),
+				 strlen(welcome_text), &work.context, FI_UINT8, FI_CSWAP_LT);
+	format_simple_msg_fetch(&fetch, &fetch_iov, result_buf, strlen(welcome_text));
+	format_simple_msg_compare(&compare, &compare_iov, compare_buf, strlen(welcome_text));
+
+	compare_atomic.ep = trig_ep;
+	compare_atomic.msg = msg;
+	compare_atomic.fetch = fetch;
+	compare_atomic.compare = compare;
+	compare_atomic.flags = 0;
+	work.op_type = FI_OP_COMPARE_ATOMIC;
+
+	work.op.compare_atomic = &compare_atomic;
+	return fi_control(&domain->fid, FI_QUEUE_WORK, &work);
+}
+
+static int tsend_trigger()
+{
+	int ret;
+	struct fi_op_tagged tagged;
+	struct fi_msg_tagged msg;
+	struct iovec msg_iov;
+
+	format_simple_msg_tagged(&msg, &msg_iov, rx_buf,
+				 strlen(write_text) + 1, &work.context, 0xCAFE);
+	tagged.ep = trig_ep;
+	tagged.msg = msg;
+	tagged.flags = 0;
+	work.op.tagged = &tagged;
+
+	work.op_type = FI_OP_TRECV;
+	ret = fi_control(&domain->fid, FI_QUEUE_WORK, &work);
+	if (ret)
+		return ret;
+
+	work.context = trig_ctx2;
+
+	format_simple_msg_tagged(&msg, &msg_iov, tx_buf + strlen(welcome_text),
+				 strlen(write_text) + 1, &work.context, 0xCAFE);
+	work.op_type = FI_OP_TSEND;
+	return fi_control(&domain->fid, FI_QUEUE_WORK, &work);
+}
+
+static int send_trigger()
+{
+	int ret;
+	struct fi_op_msg op_msg;
+	struct fi_msg msg;
+	struct iovec msg_iov;
+
+	format_simple_msg(&msg, &msg_iov, rx_buf,
+			  strlen(write_text) + 1, &work.context);
+
+	op_msg.ep = trig_ep;
+	op_msg.msg = msg;
+	op_msg.flags = 0;
+	work.op.msg = &op_msg;
+
+	work.op_type = FI_OP_RECV;
+	ret = fi_control(&domain->fid, FI_QUEUE_WORK, &work);
+	if (ret)
+		return ret;
+
+	work.context = trig_ctx2;
+
+	format_simple_msg(&msg, &msg_iov, tx_buf + strlen(welcome_text),
+			  strlen(write_text) + 1, &work.context);
+	work.op_type = FI_OP_SEND;
+	return fi_control(&domain->fid, FI_QUEUE_WORK, &work);
+}
+
+static int trigger()
+{
+	int ret;
+
+	if (opts.dst_addr) {
+		work.triggering_cntr = txcntr;
+		rx_exp = 3;
+	} else {
+		work.triggering_cntr = rxcntr;
+		rx_exp = 4;
+	}
+	work.threshold = 3;
```

Comment:
We probably need some comments about how the above constants (3, 4, 3) were chosen.   Maybe describe the high-level operational flow?